test/symmetric.jl

```diff
@@ -1199,13 +1199,64 @@ end
     end
 end
 
-@testset "asin/acos/acosh for matrix outside the real domain" begin
+@testset "asin/acos/acosh/tanh for matrix outside the real domain" begin
     M = [0 2;2 0] #eigenvalues are ±2
     for T ∈ (Float32, Float64, ComplexF32, ComplexF64)
         M2 = Hermitian(T.(M))
         @test sin(asin(M2)) ≈ M2
         @test cos(acos(M2)) ≈ M2
         @test cosh(acosh(M2)) ≈ M2
+        @test tanh(atanh(M2)) ≈ M2
+    end
+end
+
+@testset "type inference of matrix functions" begin
+    for T ∈ (Float32, Float64, ComplexF32, ComplexF64)
+        a = randn(T, 2, 2)
+        syma = Symmetric(a)
+        symtria = SymTridiagonal(syma)
+        herma = Hermitian(a)
+        #nasty functions
+        for f in (x->x^real(T)(0.3), sqrt, log, asin, acos, acosh, atanh)
+            if T <: Real
+                @test @inferred Matrix{Complex{T}} f(a) isa Matrix
```

Comment:
```suggestion
                @test @inferred(Matrix{Complex{T}}, f(a)) isa Matrix
```
Otherwise, the `@inferred` acts on the entire expression.